caas/kubernetes/provider/operator.go

```diff
@@ -295,19 +295,158 @@ func (k *kubernetesClient) validateOperatorStorage() (string, error) {
 // OperatorExists indicates if the operator for the specified
 // application exists, and whether the operator is terminating.
 func (k *kubernetesClient) OperatorExists(appName string) (caas.OperatorState, error) {
-	var result caas.OperatorState
 	operatorName := k.operatorName(appName)
+	exists, terminating, err := k.operatorStatefulSetExists(appName, operatorName)
+	if err != nil {
+		return caas.OperatorState{}, errors.Trace(err)
+	}
+	if exists || terminating {
+		if terminating {
+			logger.Tracef("operator %q exists and is terminating")
+		} else {
+			logger.Tracef("operator %q exists")
+		}
+		return caas.OperatorState{Exists: exists, Terminating: terminating}, nil
+	}
+	checks := []struct {
+		label string
+		check func(appName string, operatorName string) (bool, error)
+	}{
+		{"rbac", k.operatorRBACResourcesRemaining},
+		{"config map", k.operatorConfigMapExists},
+		{"configurations config map", k.operatorConfigurationsConfigMapExists},
+		{"service", k.operatorServiceExists},
+		{"secret", k.operatorSecretExists},
+		{"deployment", k.operatorDeploymentExists},
+		{"pods", k.operatorPodExists},
+	}
+	for _, c := range checks {
+		exists, err := c.check(appName, operatorName)
+		if err != nil {
+			return caas.OperatorState{}, errors.Annotatef(err, "%s resource check", c.label)
+		}
+		if exists {
+			logger.Debugf("operator %q exists and is terminating due to dangling %s resource(s)", c.label)
+			return caas.OperatorState{true, true}, nil
+		}
+	}
+	return caas.OperatorState{}, nil
+}
+
+func (k *kubernetesClient) operatorStatefulSetExists(appName string, operatorName string) (bool, bool, error) {
```

Comment:
either a comment or name the return params

(bool, bool) doesn't indicate what they represent